File: src/main/java/team4/footwithme/config/SchedulingConfig.java
package team4.footwithme.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.TaskScheduler;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;

@EnableScheduling
@Configuration
public class SchedulingConfig {

    public static final int POOL_SIZE = 10;
    public static final String THREAD_NAME_PREFIX = "TaskScheduler-";

    @Bean
    public TaskScheduler taskScheduler() {
        ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
        taskScheduler.setPoolSize(POOL_SIZE);
        taskScheduler.setThreadNamePrefix(THREAD_NAME_PREFIX);
        taskScheduler.initialize();
        return taskScheduler;
    }
}

File: src/main/java/team4/footwithme/config/SchedulingConfig.kt
package team4.footwithme.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler

@EnableScheduling
@Configuration
class SchedulingConfig {
    @Bean
    fun taskScheduler(): ThreadPoolTaskScheduler {
        val taskScheduler = ThreadPoolTaskScheduler()
        taskScheduler.poolSize = POOL_SIZE
        taskScheduler.threadNamePrefix = THREAD_NAME_PREFIX
        taskScheduler.initialize()
        return taskScheduler
    }

    companion object {
        const val POOL_SIZE: Int = 10
        const val THREAD_NAME_PREFIX: String = "TaskScheduler-"
    }
}
